python/tank/platform/engine.py

```diff
@@ -935,75 +1131,128 @@ def log_debug(self, msg):
         """
         Logs a debug message.
 
+        .. deprecated:: 0.18
+            Use :meth:`Engine.log` instead.
+
+        .. note:: Toolkit will probe for this method and use it to determine if
+                  the current engine supports the new :meth:`Engine.log` based logging
+                  or not. If you are developing an engine and want to upgrade it to
+                  use the new logging capabilities, you should remove the
+                  implementation of ``log_debug|error|info|...()`` methods and
+                  instead sublcass :meth:`Engine._emit_log_message`.
+
         :param msg: Message to log.
         """
-        log.debug(msg)
+        if not self.__has_018_logging_support() and self.__log_handler.inside_dispatch:
+            # special case: We are in legacy mode and all log messages are
```

Comment:
note: we are unlikely to ever get here - all engines afaik implement full logging support. But since it would result in an infinite recursion if we did get here, i thought i'd might just as well put in a check.